third_party/terraform/tests/resource_cloudscheduler_job_test.go

```diff
@@ -0,0 +1,195 @@
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccCloudSchedulerJob_pubsub(t *testing.T) {
+	t.Parallel()
+
+	jobResourceName := "google_cloud_scheduler_job.job"
+	pubSubJobName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	project := getTestProjectFromEnv()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudSchedulerJobDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudSchedulerJob_pubSubConfig(pubSubJobName, project),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudSchedulerJobExists(jobResourceName),
+					resource.TestCheckResourceAttr(jobResourceName, "name", pubSubJobName),
+					resource.TestCheckResourceAttr(jobResourceName, "description", "test job"),
+					resource.TestCheckResourceAttr(jobResourceName, "schedule", "*/2 * * * *"),
+					resource.TestCheckResourceAttr(jobResourceName, "time_zone", "Etc/UTC"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudSchedulerJob_http(t *testing.T) {
+	t.Parallel()
+
+	jobResourceName := "google_cloud_scheduler_job.job"
+	httpJobName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudSchedulerJobDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudSchedulerJob_httpConfig(httpJobName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudSchedulerJobExists(jobResourceName),
+					resource.TestCheckResourceAttr(jobResourceName, "name", httpJobName),
+					resource.TestCheckResourceAttr(jobResourceName, "description", "test http job"),
+					resource.TestCheckResourceAttr(jobResourceName, "schedule", "*/8 * * * *"),
+					resource.TestCheckResourceAttr(jobResourceName, "time_zone", "America/New_York"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudSchedulerJob_appEngine(t *testing.T) {
+	t.Parallel()
+
+	jobResourceName := "google_cloud_scheduler_job.job"
+	appEngineJobName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudSchedulerJobDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudSchedulerJob_appEngineConfig(appEngineJobName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudSchedulerJobExists(jobResourceName),
+					resource.TestCheckResourceAttr(jobResourceName, "name", appEngineJobName),
+					resource.TestCheckResourceAttr(jobResourceName, "description", "test app engine job"),
+					resource.TestCheckResourceAttr(jobResourceName, "schedule", "*/4 * * * *"),
+					resource.TestCheckResourceAttr(jobResourceName, "time_zone", "Europe/London"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCloudSchedulerJobExists(n string) resource.TestCheckFunc {
```

Comment:
I think we are generally choosing to move away from this check in the auto-generated tests, since the test would fail prior to this if the resource doesn't exist, so it's mostly redundant. It also removes the need to include the Go SDK clients, which we don't use for autogenerated resources. I'm not sure I want to add a dep in just for tests.

Reply:
If you really think it's necessary, we'll have to vendor it into terraform-provider-google-beta first since that's causing the build failure.